### ISSUE TITLE
[FlexAttention] torch._checks for unbacked symint support causing recompile limit hits

### ISSUE THREAD

**Issue by liangel-02** [pytorch/pytorch#189786]
```  File "/home/liangel/local/vllm/vllm/model_executor/layers/attention/attention.py", line 753, in unified_attention_with_output                                                         
      self.impl.forward(                                                                                                                                                                  
    File "/home/liangel/local/vllm/vllm/v1/attention/backends/flex_attention.py", line 1204, in forward                                                                                   
      out = flex_attention_compiled(                                                                                                                                                      
            ^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                      
    File "/home/liangel/local/pytorch/torch/_dynamo/eval_frame.py", line 1169, in compile_wrapper                                                                                         
      result = fn(*args, **kwargs)                                                                                                                                                        
               ^^^^^^^^^^^^^^^^^^^                                                                                                                                                        
    File "/home/liangel/local/pytorch/torch/_dynamo/convert_frame.py", line 2642, in __call__                                                                                             
      result = self._torchdynamo_orig_backend(                                                                                                                                            
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                            
    File "/home/liangel/local/pytorch/torch/_dynamo/convert_frame.py", line 778, in __call__                                                                                              
      result = _compile(                                                                                                                                                                  
               ^^^^^^^^^                                                                                                                                                                  
    File "/home/liangel/local/pytorch/torch/_dynamo/convert_frame.py", line 2045, in _compile                                                                                             
      raise FailOnRecompileLimitHit(                                                                                                                                                      
  torch._dynamo.exc.FailOnRecompileLimitHit: Hard failure due to fullgraph=True      
```
seeing a `FailOnRecompileLimitHit` error when i run 
```
WANDB_MODE=disabled python -m torchtitan.experiments.rl.train  --module alphabet_sort --config rl_grpo_qwen3_0_6b_flex
```

i narrowed it down to this PR https://github.com/pytorch/pytorch/pull/183838 which adds four torch._checks here:  https://github.com/pytorch/pytorch/blob/main/torch/nn/attention/flex_attention.py#L2462-L2477 this pins the query seq dim to a static value causing a recompile each time. if i comment out these four lines, the config runs fine (see below) 

```
Train | Step:  2  perf/trainer/step_time_ratio/batch: 0.00010  perf/trainer/step_time_ratio/blocking_generator_pull_model_state_dict: 0.16  perf/trainer/step_time_ratio/blocking_trainer_push_model_state_dict: 0.0080  perf/trainer/step_time_ratio/fwd_bwd: 0.83  perf/trainer/step_time_ratio/unaccounted: 0.0012  perf/trainer/tokens_per_second_full_step: 6233.9  perf/trainer/tokens_per_second_fwd_bwd: 7473.4  generator/inflight_requests_at_completion/max: 193.0  generator/inter_token_latency_ms/mean: 38.83  generator/queue_time_ms/mean: 3.19  generator/decode_time_ms/mean: 3844.6  generator/num_cached_tokens/mean: 42.0  loss/mean: 0  rollout_reward/_mean: 0  trainer/entropy/mean: 11.44  trainer/grad_norm/mean: 0  trainer/lr: 2.0e-06  bit_wise/logprob_diff/max: 0.039  
[titan] 2026-07-13 15:35:38,468 - root - INFO - [actor=<root>.<torchtitan.experiments.rl.actors.trainer.PolicyTrainer trainer{'gpus': 1/2}>] put_state_dict[model_state_dict]/cpu_staged:flatten took 0.0003s
[titan] 2026-07-13 15:35:38,469 - root - INFO - [actor=<root>.<torchtitan.experiments.rl.actors.trainer.PolicyTrainer trainer{'gpus': 1/2}>] put_state_dict[model_state_dict]/cpu_staged:cast took 0.0002s
[titan] 2026-07-13 15:35:38,470 - root - INFO - [actor=<root>.<torchtitan.experiments.rl.actors.trainer.PolicyTrainer trainer{'gpus': 0/2}>] put_state_dict[model_state_dict]/cpu_staged:flatten took 0.0003s
[titan] 2026-07-13 15:35:38,470 - root - INFO - [actor=<root>.<torchtitan.experiments.rl.actors.trainer.PolicyTrainer trainer{'gpus': 0/2}>] put_state_dict[model_state_dict]/cpu_staged:cast took 0.0002s
INFO 07-13 15:35:38 [loggers.py:273] [actor=<root>.<torchtitan.experiments.rl.actors.generator.VLLMGenerator generator{'gpus': 1/2}>] Engine 000: Avg prompt throughput: 73.1 tokens/s, Avg generation throughput: 419.6 tokens/s, Running: 1 reqs, Waiting: 0 reqs, GPU KV cache usage: 0.0%, Prefix cache hit rate: 71.1%
INFO 07-13 15:35:38 [loggers.py:273] [actor=<root>.<torchtitan.experiments.rl.actors.generator.VLLMGenerator generator{'gpus': 0/2}>] Engine 000: Avg prompt throughput: 73.1 tokens/s, Avg generation throughput: 419.6 tokens/s, Running: 1 reqs, Waiting: 0 reqs, GPU KV cache usage: 0.0%, Prefix cache hit rate: 71.1%
[titan] 2026-07-13 15:35:38,783 - root - INFO - [actor=<root>.<torchtitan.experiments.rl.actors.trainer.PolicyTrainer trainer{'gpus': 1/2}>] put_state_dict[model_state_dict]/cpu_staged:put_batch took 0.3140s (4.79 GB/s)
[titan] 2026-07-13 15:35:38,791 - root - INFO - [actor
```

should the fix be to gate these checks on unbacked symints only? cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @bobrenjc93 @aditvenk @laithsakka @oulgen @jamesjwu @aorenste @anijain2305 @masnesral @coconutruben @Chillee @drisspg @yanboliang @BoyuanFeng @howardzhang-cv @sanketpurandare @tianyu-l @wwwjn

**Comment by sanketpurandare** [comment]
@liangel-02 Can you give me a standalone repro?
